lucene/core/src/java/org/apache/lucene/store/ByteBuffersDataOutput.java

```diff
@@ -279,11 +279,12 @@ public ByteBuffersDataInput toDataInput() {
    * Copy the current content of this object into another {@link DataOutput}.
    */
   public void copyTo(DataOutput output) throws IOException {
-    for (ByteBuffer bb : toBufferList()) {
+    for (ByteBuffer bb : blocks) {
       if (bb.hasArray()) {
```

Comment:
this condition was always `false` before because toBufferList calls `asReadOnlyBuffer` on the buffers

Reply:
Good catch!